Create user home

cit280superepicness opened the pull request.

jrmullen commented on web/index.jsp:
```diff
@@ -0,0 +1,16 @@
+<%--
```
What is this file for?

jrmullen commented on out/artifacts/watchdog_war_exploded/user_home.html:
```diff
@@ -0,0 +1,40 @@
+<!DOCTYPE HTML>
```
what is this file?

jrmullen commented:
what are all of the jar files?

jrmullen commented on src/main/java/com/watchdog/User_HomeController.java:
```diff
@@ -0,0 +1,15 @@
+public class User_HomeController {
+
+    @RequestMapping("/user_home")
+    public void emptyMethod(){
```
refactor `emptyMethod()` to `userHome()` or something like that

cit280superepicness commented:
The index.jsp was automatically inserted when I added the Web development framework. With Java in an MVC architecture, files ending in .JSP are often used opposed to ones with the .HTML extension. However, I don't see why we can't use .HTML files instead of .JSP files like we have been. I left the index.jsp file there just in case someone wanted to use it (I believe that index.jsp is mapped in the web.xml file too), but I can get rid of it if needed. As far as all the imported .jar files, these are java libraries that were imported when I added the Web application support and other frameworks I mentioned in the email I sent last night. As far as the number of frameworks I added, I probably added some that we won't end up needing. I cannot say which .jar files correspond to which frameworks.  I will go ahead and refactor User_HomeController.java to give its method a more meaningful name. Sorry about that.

jrmullen commented:
What exactly is the web development framework? This was part of trying to get the footer fixed right? Why is this necessary I'm not sure I understand.

As for the jars we are not using jars for our project. We are using Maven to manage dependencies which replaces jars.

Footer was fixed once javascript was working by moving javascript files to the static directory, right? Is web development framework still necessary? I don't see where it is coming in. I think we can already make jsp files with spring if we find out we need them.

jrmullen commented:
If we need the web framework thing there is probably a maven dependency for it which can be added to the pom.xml file that will replace all of the jars and stuff. That is why maven is so powerful. All of our tools that we are using instead of getting a milliion jars we are just adding the 

```
<dependency>
       <groupId>org.springframework.boot</groupId>
       <artifactId>spring-boot-starter-thymeleaf</artifactId>
</dependency>
```

and it is automatically adding all of the stuff necessary for that tool to our project. I'm sure there is a maven block like this for the web framework if we find we need it.

cit280superepicness commented:
Yes, you are correct that the main reason I added it originally was to try to fix the JavaScript issue. However, it also creates a Web folder and a web.xml which is used to map servlets and other resources in the project. It seems though servlets can be mapped with Spring though too maybe even without the web.xml.

Yes, the JavaScript issue was fixed by moving footer.js to the static folder and adding th:src="@{footer.js}" to the <script></script> tags in user_home.html.

Looking at this more, I think that you may very well be right that we won't need the Web application framework since it seems very likely that there is a different way to do the same thing with Spring and Maven. Sorry if I mislead/created more work for the group by suggesting adding the Web application and other frameworks.

jrmullen commented:
You're good. I would say remove it for now, and if we discover that we did actually need it we can add it back in. How's that?

Never worry about mistakes or suggesting tools etc. The entire point of a pull request is to gain constructive criticism and to catch errors or other things that we may overlook while writing our code. It is always better to have more than one pair of eyes.

Never hesitate to suggest a tool because there is always a chance that there is a tool that would be useful to us but we simply have never heard of it. It is a group project and everyone's idea and opinions can and should be addressed.

jrmullen commented:
So it seems that what we will want for this merge is your user page, your footer, and your controller. Do you agree with this? Is there anything else?

cit280superepicness commented:
Agreed, removing it sounds like the best course of action for now. 

That said, thanks for being so open to suggestions for tools we can use. If find a good tool that may solve a problem we have in the future, I'll bring it up (hopefully doing a little more research to see if it is actually needed before I suggest it next time!) :)

No, I think just those three things are all that is needed for the merge. I'll fix up/revert my branch later tonight to before I added all those frameworks and jars, so the merge will be only of those three files. Then someone can approve the merge pull request if every looks good.

jrmullen commented:
Sounds good. I'll look at it and merge it tomorrow. I'll pull your changes and add the page to our demo tuesday too

cit280superepicness committed:
Grabbed latest version of master branch and added footer and user_hoem related files. If it looks good, is ready to merge with master

jrmullen commented:
Looks good. Let's merge.
